### ISSUE TITLE
CloudForms integration with Chef

### ISSUE THREAD

**Issue by Zorrom** [ramrexx/CloudForms_Chef#2]
The integration of CloudForms with chef using buttons are working fine. However when I try to integrate chef bootstrap as a post provisioning task for some reason cloudforms is not able to find my class and I get the following error

[----] E, [2017-12-22T12:01:36.265659 #10024:9af13c] ERROR -- : Q-task_id([service_template_provision_task_6393000000000079]) <AutomationEngine> Class [Integration/Chef/StateMachines] not found in MiqAeDatastore

However I have the exact path mentioned in post provisioning as per the documentation which is
/Integration/Chef/StateMachines/chef_bootstrap_${/#miq_provision.source.platform}

Kindly provide a solution on this and thanks for all the Chef Integration domains and methods :+1: 
![cloudforms automation](https://user-images.githubusercontent.com/33654256/34290630-84983cb2-e71e-11e7-8b06-5bd631dba5fb.png)

**Comment by Zorrom** [comment]
Hi,
   Now I have changed the entry in Post2 as /Integration/Chef/..../../chef_bootstrap_linux and my method got picked up. But during bootstrap check I am getting the following error with chef node name. However my VM has spun with proper IP and FQDN
![vm provision error](https://user-images.githubusercontent.com/33654256/34352940-604f0b08-ea4b-11e7-9790-52fb22096b26.PNG)
 PFA and kindly assist on the same

**Comment by ramrexx** [comment]
Hi Zorrom,
I think i see the problem. It looks like you added the path "/Integration/Chef/StateMachines/chef_bootstrap_${/#miq_provision.source.platform}" to the wrong instance.

Instead of "/ Service / Provisioning / StateMachines / CatalogItemInitialization" put it in the "/ Infrastructure / VM / Provisioning / StateMachines / VMProvision_VM / Provision VM from Template" and let me know if that helps.

**Comment by Zorrom** [comment]
Yes, I tried changing the path as you mentioned and tried spinning up the VM but it failed at the initial stage stating as Reqeuest completed with errors.
I checked the logs and found that its failing at the part where its looking for miq_provisision object not provided. PFA for the same.

My self service catalog only accepts the following inputs which are as follows

Service name  - service_name
Vm name        -  vm_name
Number of CPUs - option_0_cores_per_socket
VM Memory      - option_0_vm_memory
IP Address          - ip_addr
Host Name         - linux_host_name

Let me know if these inputs conflict with it by any chance.

Thanks and Regards,
Azhar

**Comment by Zorrom** [comment]
Yes, I tried changing the path as you mentioned and tried spinning up the VM but it failed at the initial stage stating as Reqeuest completed with errors.
I checked the logs and found that its failing at the part where its looking for miq_provisision object not provided. PFA for the same.

My self service catalog only accepts the following inputs which are as follows

Service name  - service_name
Vm name        -  vm_name
Number of CPUs - option_0_cores_per_socket
VM Memory      - option_0_vm_memory
IP Address          - ip_addr
Host Name         - linux_host_name

Let me know if these inputs conflict with it by any chance. 


![error message](https://user-images.githubusercontent.com/33654256/34509919-943df7e6-f075-11e7-90ac-663d4ec89ee8.PNG)
